mcapi/mcapi.proto

```diff
@@ -0,0 +1,437 @@
+
+
+syntax = "proto3";
+
+package report;
+
+// placeholders
+message Interval {
+  string start_time = 1;
+  string end_time = 2;
+}
+
+// The below assumes a set of defaults for each deployment's
+// event templates.
+// To support this each event template will need to be tagged with
+// the following:
+// 1. whether it is a media type event template and if so what the
+// media type is (e.g. video, display, etc.)
+// 2. whether it represents a common set of event templates that
+// span media types.
+// 3. the deployment/market it is associated with
+
+// Upon receiving a group of reporting sets, the API will be able to
+// infer what event templates are involved and use that information
+// to find the event tempalte definitions and apply defaults as needed.
+// This is how the market measurement policy will be enforced.
+
+// A set of defaults for the API will be specified such that default
+// impression filters and other things can be determined. This mechanism
+// will also enforce those aspects of the report that must exist.
+// For example the Origin measurement policy requires MRC and AMI
+// for Video in all reports.
+// Details on the full set of defaults and the mechanisms for
+// handling this are TBD.
+
+// A Basic Report is composed of the following:
+// At the top level the following are specified:
+// 1. the report name
+// 2. the reporting start and end times
+// 3. media type specific filtering criteria (e.g. ami, mrc, custom).
+//    defaults will exist for these and the measurement policy will be
+//    enforced with respect to what may/must be reported on
+// 4. a set of "page definitions" that specify which metrics to compute
+// for which combination of reporting sets
+// 5. a set of pages, one for each definition, that contain the results
+//
+// *** A note on the use of ReportingSets:
+// The API will only work with primitive Reporting Sets or
+// composite Reporting Sets that are composed only of unions and whose
+// constituent Reporting Sets have no filters.
+// Additionally all Event Groups in the reporting set must conform to a
```

Comment:
We want to relax this and allow reporting sets to have EGs with hetergenous MediaType. Thus you could specify one RS per EDP (or per creative if you wanted to report on creatives x-EDP).